core/com.dexels.navajo.function/src/com/dexels/navajo/functions/Max.java

```diff
@@ -86,20 +93,27 @@ private Object max(Object a, Object b)
         }
         else
         {
-            if ((b instanceof Date))
+            if ( b != null && ( b instanceof Date || b instanceof ClockTime ) )
```

Comment:
you can leave out the null check, see: https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof

Reply:
The null check is not there for the instanceof. Right now if you enter a Date and a null value you end up at the coded exception "Cannot compare a date with a non-date". But it should be possible to compare these with the Date always winning, both at Min and Max.